src/vm/methodtable.cpp

```diff
@@ -1398,10 +1388,12 @@ BOOL MethodTable::IsEquivalentTo_WorkerInner(MethodTable *pOtherMT COMMA_INDEBUG
     }
     CONTRACTL_END;
 
+    TypeEquivalenceHashTable *typeEquTable = nullptr;
     AppDomain *pDomain = GetAppDomain();
```

Comment:
I guess this will need to be update for unloading ... assuming we make COM types unloadable.

cc @janvorli

Reply:
At the moment, collectible types are not cached in the equivalence cache, see the code at the end of this function. So we should be safe. But it would be good to enable caching even for collectible types. I've created #21277 to track that.